### PR TITLE
INTYGFV-14460: Added possibility to search for patient

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

**Review by perfli — CHANGES_REQUESTED** [review]

File: packages/webcert/src/components/header/WebcertHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import WebcertHeaderUser from './WebcertHeaderUser'
+import WebcertHeaderUnit from './WebcertHeaderUnit'
+import SystemBanners from '../notification/SystemBanners'
+import AboutWebcertModalContent from '../../feature/certificate/Modals/AboutWebcertModalContent'
+
+interface Props {
+  isEmpty?: boolean
+}
+
+const WebcertHeader: React.FC<Props> = ({ isEmpty = false }) => {
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply packages/webcert/src/components/header/WebcertHeader.tsx:13]: The other header components have Webcert before so I was following the same pattern

File: packages/webcert/src/components/patient/PatientSearch.test.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  )
+}
+
+describe('PatientSearch', () => {
+  it('should render component', () => {
+    renderComponent()
+  })
+
+  describe('Input', () => {
+    it('should allow user to type patient id', () => {
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply packages/webcert/src/components/patient/PatientSearch.test.tsx:24]: No it tests the connection between user input and value for component. The textinput has logic deciding the value and it doesn't just set the input as its value.

File: packages/webcert/src/page/CreateCertificatePage.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+  useEffect(() => {
+    ReactTooltip.hide()
+  }, [patient])
+
+  useEffect(() => {
+    if (patientId) {
+      dispatch(getPatient({ patientId: patientId, history: history }))
+    }
+  }, [history, dispatch, patientId])
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply packages/webcert/src/page/CreateCertificatePage.tsx:36]: This was suggested from IntelliJ. It gives a warning if I don't add it.

File: packages/webcert/src/App.tsx [hunk 4]
```diff
@@ -44,13 +45,21 @@ function App(): JSX.Element {
...
-            <ErrorBoundary fallbackRender={({ error }) => <>Ett fel har inträffat: error.message</>} onError={onError}>
+            <ErrorBoundary fallbackRender={({ error }) => <>Ett fel har inträffat: {error.message}</>} onError={onError}>
               <CertificatePage />
             </ErrorBoundary>
           )}
         />
         <Route path="/welcome" render={() => <Welcome />} />
         <Route path={'/error'} render={() => <ErrorPage />} />
+        <Route
+          path={'/create/:patientId?'}
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply packages/webcert/src/App.tsx:56]: This is how it has been before. Search/create will be accessible from the same tab later so I think it makes sense to have them in the same route.

File: packages/webcert/src/store/uiReducers.ts [hunk 5]
```diff
@@ -8,6 +8,7 @@ import questionReducer from './question/questionReducer'
 import icfReducer from './icf/icfReducer'
 import sessionReducer from './session/sessionReducer'
 import errorReducer from './error/errorReducer'
+import patientReducer from './patient/patientReducer'
 
 export default combineReducers({
   uiCertificate: certificateReducer,
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply packages/webcert/src/store/uiReducers.ts:14]: This is how we have done it from the beginning. For new states we should follow the old pattern.

File: packages/webcert/src/App.tsx [hunk 6]
```diff
@@ -44,13 +45,21 @@ function App(): JSX.Element {
         <Route
           path="/certificate/:certificateId"
           render={() => (
-            <ErrorBoundary fallbackRender={({ error }) => <>Ett fel har inträffat: error.message</>} onError={onError}>
+            <ErrorBoundary fallbackRender={({ error }) => <>Ett fel har inträffat: {error.message}</>} onError={onError}>
```
[the remark this answers is not in the document]
Comment (mhornfeldt) [reply packages/webcert/src/App.tsx:48]: I don't know I was following the pattern from before. If we put it one level up then the errorpage will have it too.

[the remark this answers is not in the document]
Comment (perfli) [reply packages/webcert/src/App.tsx:48]: I don't think that would be a bad thing, even if `ErrorPage` never generates an unhandled error.

[the remark this answers is not in the document]
Comment (mhornfeldt) [reply packages/webcert/src/App.tsx]: Okay I've moved it now

**Review by HelenaEdblad — APPROVED** [review]

**Pull request merged**